content/articles/how-to-handle-file-uploads-with-flask/index.md

```diff
@@ -0,0 +1,255 @@
+    app.run(debug=True)
+```
+from the code above we start by importing FlaskForm from flask_wtf and FileField from wtforms. Then we created a class for our form as ```Myform``` image is the file field our image files will be saved to, then we call our Form class in our index function. We changed our render to render template, this is also a flask library used for rendering HTML templates from the code we rendered ```index.html```. Now let us create the HTML template we are rendering. update the code to the one below to do that 
+```python
+!doctype html>
+<html>
+  <head>
+    <title>File Upload</title>
+  </head>
+  <body>
+
+    <form action="/" method="POST" enctype="multipart/form-data">
+        {{ form.csrf_token }}
+        {{ form.image }}
+        <button type="submit">upload</button>
+    </form>
+  </body>
+</html>
+```
+From the above code, our form takes a method ```POST``` because we will be posting a file the ```csrf_token``` is a built-in function that handles security for us, then we call our form field we created in our ```Form Class``` using ```form.image```
+
+now we can run our app using ```python app.py``` if everything is correct you will get a runtime error like shown in the below image.
+
+['image']
+
+this error occurs whenever you try to use a csrf_token without adding a secret_key to your project file. let's add a secret key to our code. update your code to the one below
+
+```python 
+from flask import Flask, render_template
+from flask_wtf import FlaskForm
+from wtforms import FileField
+
+app = Flask(__name__)
+
+app.config['SECRET_KEY'] = 'mysecretkey'
+
+class MyForm(FlaskForm):
+    image = FileField('image')
+
+@app.route('/')
+def index():
+    form = MyForm()
+    return render_template('index.html')
+
+if __name__==('__main__'):
+    app.run(debug=True)
+```
+the Secret_key can be anything you want.
+
+let's update our code to the one below.
+```python
+from flask import Flask, render_template
+from flask_wtf import FlaskForm
+from wtforms import FileField
+
+app = Flask(__name__)
+
+app.config['SECRET_KEY'] = 'mysecretkey'
+
+class MyForm(FlaskForm):
+    image = FileField('image')
+
+@app.route('/')
+def index():
+    form = MyForm()
+    return render_template('index.html', form = form)
+
+if __name__==('__main__'):
+    app.run(debug=True)
+```
+our page should now look as shown in the picture below
+
+[image file upload page]
+
+from the code above ```form=form``` is parsed so that our form can be displayed on our HTML page. if we try to upload an image we will encounter another error as shown below
+
+[image of not allowed error]
+
+this error is often coursed when we don't specify a method to our route. To solve this we will add the code below to our route.
+
+```python
+@app.route('/', methods=['GET', 'POST'])
+```
+after adding the above code, our upload will work but it won't be saved because we didn't give it a path to save to. this is where flask uploads come into play. lets import flask-uploads
```

Comment:
```suggestion
After adding the above code, our upload will work but it won't be saved because we didn't give it a path to save to. this is where `flask uploads` come into play. Let's import `flask-uploads` using the command:
```